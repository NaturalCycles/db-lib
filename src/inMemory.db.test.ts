import { InMemoryDB } from './inMemory.db'
import { runCommonDaoTest } from './testing/daoTest'
import { runCommonDBTest } from './testing/dbTest'

const db = new InMemoryDB()

describe('runCommonDBTest', () => runCommonDBTest(db))

describe('runCommonDaoTest', () => runCommonDaoTest(db))
