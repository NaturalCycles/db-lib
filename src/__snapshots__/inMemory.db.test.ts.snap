// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`save with id, load, delete 1`] = `
Object {
  "a": "aa",
  "b": "bb",
  "c": 1,
  "created": 1529539200,
  "id": "randomid123",
  "updated": 1529539200,
}
`;

exports[`save, load 1`] = `
Object {
  "a": "aa",
  "b": "bb",
  "c": 1,
  "created": 1529539200,
  "id": "randomDatastoreService1",
  "updated": 1529539200,
}
`;

exports[`streamQuery 1`] = `
Array [
  Object {
    "a": "aa",
    "b": "b1",
    "created": 1529539200,
    "id": "id1",
    "updated": 1529539200,
  },
  Object {
    "a": "aa",
    "b": "b2",
    "created": 1529539200,
    "id": "id2",
    "updated": 1529539200,
  },
]
`;
